### PR TITLE
Hide the plugin banner on non-wpcom-connected users or agency-managed users

### PR DESCRIPTION
**Pull request by okmttdhr**
Fixes https://github.com/Automattic/dotcom-forge/issues/8422

## Proposed changes:

This PR changes to hide the plugin banner on non-wpcom-connected users or agency-managed users.

| before | after  |
|--------|--------|
| Screenshot 2024-07-26 at 15 08 52 | Screenshot 2024-07-26 at 15 07 50 | 

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Prepare an Atomic Classic site
* Patch mu-wpcom change via https://github.com/Automattic/jetpack/pull/38532#issuecomment-2249533772
* Patch wpsomsh change by using `pnpm jetpack rsync wpcomsh YOURSITE@sftp.wp.com:htdocs/wp-content/mu-plugins/wpcoms`
	* Patching WordPress.com Site Helper by using Jetpack Beta errored out for me 
* Go to "Plugins > Add New Plugin"
	* Observe the banner is displayed
* Set `is_fully_managed_agency_site=1` option on the site
	* Observe the banner is not displayed
	* Revert `is_fully_managed_agency_site` to `0`
* Add a new user via "Users > Add New User" (don't connect it to a wpcom account)
	* Login with that added user
	* Go to "Plugins > Add New Plugin"
	* Observe the banner is not displayed
	* Connect the account with wpcom (e.g., via Jetpack > My Jerpack > "Some features require authentication. Sign in")
	* Go to "Plugins > Add New Plugin"
	* Observe the banner is displayed

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-plugins/wpcom-plugins.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	if ( isset( $_GET['tab'] ) && 'favorites' === $_GET['tab'] ) {
+		// no banner on the favorites tab, it's a bit overbearing given they presumably want
+		// something specific.
+		return;
+	}
+
+	// No banner for agency-managed sites.
+	if ( ! empty( get_option( 'is_fully_managed_agency_site' ) ) ) {
+		return;
+	}
```
Comment (fushar) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-plugins/wpcom-plugins.php]: This check is no longer required, as we already gate it inside `load_wpcom_user_features()` 😄

**Review by lupus2k — DISMISSED** [review]
Looks great :heavy_check_mark: 

Tested the Banner link and points to the correct page on WPcom.

Admin user + is_fully_managed_agency_site = off
image

Admin user + is_fully_managed_agency_site = on
image

New user (no WP account)
image

New user (with WP connected account) + is_fully_managed_agency_site = off
image

New user (with WP connected account) + is_fully_managed_agency_site = on
image

**Comment by okmttdhr** [comment]
Could I get approval once again, @fushar @lupus2k?

File: projects/plugins/wpcomsh/wpcomsh.php [hunk 2]
```diff
@@ -139,8 +139,6 @@
 
 require_once __DIR__ . '/wpcom-migration-helpers/site-migration-helpers.php';
 
-require_once __DIR__ . '/wpcom-plugins/plugins.php';
```
Comment (fushar) [inline projects/plugins/wpcomsh/wpcomsh.php:142]: I assume that the old code will be deleted later.

**Review by fushar — DISMISSED** [review]
Tested. I got the same results as @lupus2k's above.

**Review by lupus2k — APPROVED** [review]
Tested after the last changes and got the same results as [here](https://github.com/Automattic/jetpack/pull/38532#pullrequestreview-2202608531)

**Pull request merged**